packages/11ty/_lib/urlPathJoin/index.js

```diff
@@ -0,0 +1,23 @@
+import { posix as path } from 'node:path'
+/**
+ * @function urlPathJoin
+ *
+ * @argument {String} url
+ * @argument {Iterable} segs
+ *
+ * Parses `url` and safely joins all of `segs` to its pathname
+ *
+ * @return {String}
+ *
+ **/
+
+const urlPathJoin = (url, ...segs) => {
+  const base = new URL(url)
+  const basePath = base.pathname
+
+  base.pathname = path.join(basePath, ...segs)
+
+  return base.href
+}
+
+export default urlPathJoin
```

Comment:
⚙️ @cbutcosk the following may be a more idiomatic use of the node api
```suggestion
import path from 'node:path'

/**
 * @function urlPathJoin
 *
 * @argument {String} baseUrl
 * @argument {Iterable} segments
 *
 * Parses `url` and safely joins all of `segs` to its pathname
 *
 * @return {String}
 */
const urlPathJoin = (baseUrl, ...segments) => {
  const url = new URL(path.posix.join(...segments), baseUrl)
  return url.toString()
}

export default urlPathJoin
```

Reply:
@mphstudios The variable renames and shift away from renamed imports are 👍 (commits are ready to push) but relying on the relative URL rebasing of `new URL(input, base)` has some unpleasant corner cases for callers. For the pathname to join successfully `base` must end in '/' and `input` must not be an absolute path. Neither of these are the case for callers here, who in practice are expecting path.join()-style path joins, just with a base URL at the front (path.posix.join() normalizes the URL protocol's '//' to '/' in those cases).